### PR TITLE
if CIME_DRIVER=nuopc add pylint check for nuopc files

### PR DESCRIPTION
**Pull request by jedwards4b**
Add the external files in the nuopc directory to code_checker if the CIME_DRIVER=nuopc 

Test suite: CIME_DRIVER=nuopc scripts_regression_tests.py
Test baseline: 
Test namelist changes: 
Test status: bit for bit
Fixes [CIME Github issue #]

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

File: scripts/lib/CIME/code_checker.py [hunk 1]
```diff
@@ -92,7 +95,8 @@ def check_code(files, num_procs=10, interactive=False):
     # is a valid file, if not, we search the repo for a file with similar name.
     files_to_check = []
     if files:
-        repo_files = run_cmd_no_fail('git ls-files', from_dir=get_cime_root(), verbose=False).splitlines()
+        repo_files = get_all_checkable_files()
+#        repo_files = run_cmd_no_fail('git ls-files', from_dir=get_cime_root(), verbose=False).splitlines()
```
Comment (jedwards4b) [inline scripts/lib/CIME/code_checker.py]: @jgfouca looking for feedback on this change - if it's okay I'll remove the commented code.

File: scripts/lib/CIME/code_checker.py [hunk 2]
```diff
@@ -75,9 +75,12 @@ def get_all_checkable_files():
 ###############################################################################
     cimeroot = get_cime_root()
     all_git_files = run_cmd_no_fail("git ls-files", from_dir=cimeroot, verbose=False).splitlines()
-
+    if get_cime_default_driver() == "nuopc":
```
Comment (jgfouca) [inline scripts/lib/CIME/code_checker.py]: Curious as to why this was needed. all_git_files are computed from cimeroot, so wouldn't that include the nuopc subdirectory?

Comment (jedwards4b) [reply scripts/lib/CIME/code_checker.py]: Because the nuopc subdirectory is no longer part of the same git repo.

Comment (jgfouca) [reply scripts/lib/CIME/code_checker.py]: Ah, ok.

File: scripts/lib/CIME/code_checker.py (hunk 1 above)
Comment (jgfouca) [reply scripts/lib/CIME/code_checker.py]: It looks ok. I can't remember why I used ls-files instead of checkable_files.

**Pull request merged**